spring-boot/src/main/java/com/sentrydemos/springboot/DatabaseHelper.java

```diff
@@ -82,22 +89,47 @@ public String mapAllProductsJoin(ISpan transaction) {
 			ja.put(jsonItemObject);
 		}
 		
-		sqlSpan.setTag("totalProducts", String.valueOf(allItems.size()));
-		
-		sqlSpan.finish();
-		
 		return ja.toString();
 	}
+	
+	public List<Item> getAllItemsRowMapper(List<Item> items, ISpan transaction) {
+		Map<Integer, List<Review>> reviewsMap = new HashMap<>();
+		String sql = "SELECT reviews.id, products.id AS productid, reviews.rating, reviews.customerId, reviews.description, reviews.created FROM reviews INNER JOIN products ON reviews.productId = products.id";
+		ISpan sqlSpan = transaction.startChild("db", sql);
+		
+		List<Map<String, Object>> resultList = jdbcTemplate.queryForList(sql);
+		for (int i = 0; i < resultList.size(); i++) {
+			Map<String, Object> m = resultList.get(i);
+			
+			//customerId and description are null. TODO: Set SQL table to prevent null values
+			String desc = "";
+			if (m.containsKey("description") && m.get("description") != null) {
+				desc = String.valueOf(m.get("description"));
+			}
+			int custId = 0;
+			if (m.containsKey("customerId") && m.get("customerId") != null) {
+				custId = (int)m.get("customerId");
+			}
 
-	public List<Review> mapAllReviewsJoin(int productId, ISpan span) {
-		String sql = "SELECT * FROM reviews WHERE productId = " + String.valueOf(productId);
-
-		ISpan sqlSpan = span.startChild("db", sql);
-		List<Review> reviews = jdbcTemplate.query(sql, (rs, rowNum) -> new Review(rs.getInt("id"), rs.getInt("productid"),
-				rs.getInt("rating"), rs.getInt("customerid"), rs.getString("description"), rs.getString("created")));
-		//TODO: sqlSpan.setData() on reviews. setData() currently unavailable
+        	Review r = new Review((int)m.get("id"), (int)m.get("productid"),
+    				(int)m.get("rating"), custId, desc, m.get("created").toString());
+        	
+        	if (reviewsMap.containsKey((int)m.get("productid"))) {
+        		reviewsMap.get((int)m.get("productid")).add(r);
+        	} else {
+        		List<Review> rl = new ArrayList<Review>();
+        		rl.add(r);
+        		reviewsMap.put((int)m.get("productid"), rl);
+        	}
+		}
+		
+		for (Item i : items) {
+			i.setReviews(reviewsMap.get(i.getId()));
+		}
+		
 		sqlSpan.finish();
-		return reviews; 
+		
+		return items;
```

Comment:
your function definition says it returns a `List<Review>` and not List of Items. Is this okay? Maybe something else got renamed/refactored to support this. Your Tests (transactions you linked) looked good, so if you got what you need, it looks good. @sdzhong

Reply:
Yep. This is okay. The (Items class contains a Review list)[https://github.com/sentry-demos/application-monitoring/blob/master/spring-boot/src/main/java/com/sentrydemos/springboot/Item.java#L21]; the items in `return items` contain their respective reviews.